### PR TITLE
kconfig: Replace defconfig singe-symbol 'if's with 'depends on'

### PR DESCRIPTION
**Pull request by ulfalizer**
Same deal as in commit eddd98f ("kconfig: Replace some single-symbol
'if's with 'depends on'"), for all symbols defined within defconfig
files. See that commit for an explanation.

Maybe 'if's were used originally to mirror the 'if's in the main Kconfig
files, and then it got copied around by people assuming 'if' must work
differently from 'depends on'. It doesn't match in every spot at least.
Better to keep it simple and just consistently use 'depends on' when
it's a single symbol/choice I think. Helps reinforce that 'if' isn't
magic too.

Verified by printing all Kconfig menu nodes (symbols, choices, menus,
etc.) before and after the change and diffing (should show no
difference).

### CODE REVIEW

**Review by dcpleung — APPROVED** [review]

**Comment by ulfalizer** [comment]
Another one that'll always time out.

**Review by aescolar — APPROVED** [review]
No complaints in my files

**Review by jfischer-no — APPROVED** [review]

File: boards/arm/nucleo_wb55rg/Kconfig.defconfig [hunk 1]
```diff
@@ -44,34 +34,21 @@ config I2C_3
...
 
 config ADC_1
 	default y
-
-endif # ADC
-
-if BT_DEBUG_MONITOR
+	depends on ADC
 
 config LPUART_1
```
Comment (erwango) [inline boards/arm/nucleo_wb55rg/Kconfig.defconfig]: This symbol was defined under 2 different if, and now is defined twice with 2 different "depends on".
Is that strictly equivalent ?

File: boards/arm/stm32f411e_disco/Kconfig.defconfig [hunk 2]
```diff
@@ -13,13 +13,9 @@ if UART_CONSOLE
 config UART_2
 	default y
```
Comment (erwango) [inline boards/arm/stm32f411e_disco/Kconfig.defconfig]: `depends on UART_CONSOLE`

Initial PWM definition was wrongly placed

File: boards/arm/stm32h747i_disco/Kconfig.defconfig [hunk 3]
```diff
@@ -38,28 +38,21 @@ config CLOCK_STM32_D3PPRE
...
 	default y
 
-# Dual core boot configuration
-if STM32H7_DUAL_CORE
 # Boot method is selected in common file to avoid desync issues
 choice STM32H7_DUAL_CORE_BOOT
 	# Use out of the box config by default
 	# default STM32H7_BOOT_CM4_CM7
 	default STM32H7_BOOT_CM7_CM4GATED
+	# Dual core boot configuration
```
Comment (erwango) [inline boards/arm/stm32h747i_disco/Kconfig.defconfig]: Please don't move the comment

**Review by erwango — CHANGES_REQUESTED** [review]
A question and a request.

File: boards/arm/nucleo_wb55rg/Kconfig.defconfig (hunk 1 above)
Comment (ulfalizer) [reply boards/arm/nucleo_wb55rg/Kconfig.defconfig]: Yup, equivalent. `if` just adds a `depends on` to all symbols within it. `depends on` is the only thing that "really" exists. `if` is just sugar.

File: boards/arm/stm32f411e_disco/Kconfig.defconfig (hunk 2 above)
Comment (ulfalizer) [reply boards/arm/stm32f411e_disco/Kconfig.defconfig]: Fixed

File: boards/arm/stm32h747i_disco/Kconfig.defconfig (hunk 3 above)
Comment (ulfalizer) [reply boards/arm/stm32h747i_disco/Kconfig.defconfig]: Moved back. Thought it was about the symbol in the `if`.

**Review by vanti — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Comment by galak** [comment]
Merging this a CI will always timeout due to large number of changes.

**Pull request merged**